reader_concurrency_semaphore.cc

```diff
@@ -495,6 +495,10 @@ class reader_permit::impl
         return bool(_ex);
     }
 
+    const std::exception_ptr& get_abort_exception() const {
```

Comment:
why not mark this one noexcept as well?
although today it's inlined if it ever moves outside this file I'd expect this interface to be noexcept.

Reply:
I thought I did, maybe I only marked the external one (reader_permit).

Reply:
Fixed in v3.